pre-commit.sh

```diff
@@ -57,13 +57,7 @@ fi
 # 1. cargo clippy
 
 echo -e "$(GREEN INFO): cargo clippy ..."
-
-# Cargo clippy always return exit code 0, and `tee` doesn't work.
-# So let's just run cargo clippy.
-cargo clippy --all-targets --workspace --features avro,pyarrow -- -D warnings
-pushd datafusion-cli
-cargo clippy --all-targets --all-features -- -D warnings
-popd
+./ci/scripts/rust_clippy.sh
```

Comment:
I also double checked that this is the same as used in CI: https://github.com/apache/datafusion/blob/bcf715c892f74d48bdbef54ac7165358be6fb741/.github/workflows/rust.yml#L486

👍

Reply:
and in `dev/rust_lint.sh`